gpu/src/cuda/shaders/fields/stark256.cu

```diff
@@ -0,0 +1,21 @@
+#include "./fp_u256.cuh"
+#include "../fft/fft.cuh"
+
+namespace p256
+{
+    // StarkWare field for Cairo
+    // P =
+    // 3618502788666131213697322783095070105623107215331596699973092056135872020481
+    using Fp = Fp256<
+        /* =N **/ /*u256(*/ 576460752303423505, 0, 0, 1 /*)*/,
+        /* =R_SQUARED **/ /*u256(*/ 576413109808302096, 18446744073700081664,
+        5151653887, 18446741271209837569 /*)*/,
+        /* =N_PRIME **/ /*u256(*/ 576460752303423504, 18446744073709551615,
+        18446744073709551615, 18446744073709551615 /*)*/
+        >;
+} // namespace p256
+
+extern "C" __global__ void radix2_dit_butterfly(p256::Fp *input, const p256::Fp *twiddles)
```

Comment:
in metal I think the function has the field name appended.

Reply:
Yes, but in CUDA that makes things more difficult. The `load_ptx` function receives a module name and a list of function names. The function names have a _static_ lifetime bound (i.e. are `&'static str`), so they can't be resolved at runtime. I bypassed it by using the module name (that doesn't have lifetime bounds) to differentiate between each implementation, but in turn I need the function names to be fixed and also to have each implementation in a different PTX.